cmd/kube-proxy/app/server_others.go

```diff
@@ -164,6 +164,56 @@ func newProxyServer(
 			healthzServer,
 			config.NodePortAddresses,
 		)
+
+		if utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack) {
+			klog.V(0).Info("creating dualStackProxier for iptables.")
+
+			// Create iptables handlers for both families, one is already created
+			// Always ordered as IPv4, IPv6
```

Comment:
I think this ordering constraint will be surprising for future code changes, and any mistakes won't be noticed until runtime :(

Can we keep this in arbitrary order here, and move the `if IsIpv6()` check into `NewDualstackProxier` where the interfaces are actually used instead?  (It looks like we'll need to "duplicate" not just "move", but the key point is that the ordering is not important between this code and the code elsewhere in NewDualstackProxier)

An alternative might be inlining NewDualStackProxier here (ie: construct the v4/v6 -specific proxiers and wrap in MetaProxier here and remove the need to ever have a generic `ipt[2]` intermediate)

Reply:
To be fair, this is copying the ipvs logic.  If we want to raise the bar, we should do so for both proxiers.

Perhaps, instead of `[2]utiliptables.Interface` it should be a `map[core.IPFamily]utiliptables.Interface`.  Remove order from the equasion, since that seems to be meaningless anyway (and assumed ordering is bound to explode on us some day).